packages/valory/contracts/dynamic_contribution/contract.py

```diff
@@ -106,11 +106,24 @@ def get_all_erc721_transfers(
         """
         ledger_api = cast(EthereumApi, ledger_api)
         factory_contract = cls.get_instance(ledger_api, contract_address)
-        entries = factory_contract.events.Transfer.createFilter(
-            fromBlock=from_block,
-            toBlock=to_block,
-            argument_filters={"from": from_address},
-        ).get_all_entries()  # limited to 10k entries for now: https://github.com/valory-xyz/contribution-service/issues/13
+
+        # Avoid parsing too many blocks at a time. This might take too long and
+        # the connection could time out.
+        MAX_BLOCKS = 300000
+        to_block = (
+            ledger_api.api.eth.get_block_number() if to_block == "latest" else to_block
+        )
+        ranges = list(range(from_block, to_block, MAX_BLOCKS)) + [to_block]
```

Comment:
So we are doing 30000 RPC calls?

Reply:
No, 1 call that retrieves events for 300k blocks. After some testing I think that's the max amount we should get.